File: server/net/Models/Countdown.cs
using System;

namespace Daysuntil.Models
{
    public class Countdown
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public required DateTime DateTime { get; set; }
        public bool IsPublic { get; set; }
        public int? CategoryId { get; set; }
        public required int UserId { get; set; }

        public Category? Category { get; set; }
        public User? User { get; set; }
    }
}

File: server/laravel/app/Models/Countdown.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\User;
use App\Models\Category;
use OpenApi\Attributes as OA;

#[OA\Schema(schema: "Countdown", description: "Countdown model")]
class Countdown extends Model
{
    use HasFactory;

    protected $fillable = ['name', 'datetime', 'is_public', 'user_id', 'category_id'];

    /**
     * Get the user that owns the countdown.
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the category associated with the countdown.
     */
    public function category()
    {
        return $this->belongsTo(Category::class);
    }
}